<launch>
    <arg name="output_dir"/>
    <arg name="urdf_file" default="$(find jaco_description)/urdf/jaco_robot.urdf"/>

    # factor to scale the model by 
    <arg name="scale_factor" default="1"/>

    # An axis and angle (degrees) can be specified which will transform *all*
    # visuals (not links, but their visuals!) within their local coordinate system.
    # This can be used to correct transformation errors which may have been 
    # introduced in converting meshes from one format to the other, losing orientation information
    # For example, .dae has an "up vector" definition which may have been ignored.
    <arg name="visual_corr_axis_x" default="1"/>
    <arg name="visual_corr_axis_y" default="0"/>
    <arg name="visual_corr_axis_z" default="0"/>
    <arg name="visual_corr_axis_angle" default="0"/>

	<node name="urdf2inventor" pkg="urdf2inventor" type="urdf2inventor_node" respawn="false"
        output="screen" args="$(arg urdf_file) $(arg output_dir)">
		<param name="scale_factor" value="$(arg scale_factor)"/>
        <param name="visual_corr_axis_x" value="$(arg visual_corr_axis_x)"/>
        <param name="visual_corr_axis_y" value="$(arg visual_corr_axis_y)"/>
        <param name="visual_corr_axis_z" value="$(arg visual_corr_axis_z)"/>
        <param name="visual_corr_axis_angle" value="$(arg visual_corr_axis_angle)"/>
    </node>
</launch>
